artichoke-backend/src/extn/core/integer/mod.rs

```diff
@@ -98,10 +98,13 @@ impl Integer {
 
     pub fn chr(self, interp: &mut Artichoke, encoding: Option<Value>) -> Result<Vec<u8>, Error> {
         if let Some(encoding) = encoding {
-            let mut message = b"encoding parameter of Integer#chr (given ".to_vec();
+            let mut message = spinoso_string::String::ascii(b"encoding parameter of Integer#chr (given ".to_vec());
             message.extend(encoding.inspect(interp));
             message.extend(b") not supported");
-            Err(NotImplementedError::from(message).into())
+            match spinoso_string::String::alloc_value(message, interp) {
+                Ok(msg) => Ok(msg.inspect(interp)),
+                Err(_) => Err(NotImplementedError::from(message).into()),
+            }
```

Comment:
This code constructs a `Vec<u8>` which is passed directly to an exception constructor. It does not need to go through a spinoso string.